Content.Shared/Vehicle/SharedVehicleSystem.cs

```diff
@@ -129,26 +131,71 @@ private void OnVehicleStartup(EntityUid uid, VehicleComponent component, Compone
     /// change its draw depth. Vehicles can choose between special drawdetph
     /// when facing north or south. East and west are easy.
     /// </summary>
-    protected int GetDrawDepth(TransformComponent xform, bool northOnly)
+    protected int GetDrawDepth(TransformComponent xform, VehicleComponent component, double cameraDegrees)
     {
-        // TODO: I can't even
-        if (northOnly)
+        var cameraAngle = GetAngle(cameraDegrees + 90);
+
+        var itemAngle = GetAngle(xform.LocalRotation.Degrees + cameraAngle + 90);
+
+        var itemDirection = string.Empty;
+        var cameraDirection = cameraAngle switch
         {
-            return xform.LocalRotation.Degrees switch
-            {
-                < 135f => (int) DrawDepth.DrawDepth.Doors,
-                <= 225f => (int) DrawDepth.DrawDepth.WallMountedItems,
-                _ => 5
-            };
+            >= 0 and < 90 => "Right",
+            >= 90 and < 180 => "Down",
+            >= 180 and < 270 => "Left",
+            _ => "Up"
+        };
+
+        switch (cameraDirection)
+        {
+            case "Down":
+            case "Up":
+                itemDirection = itemAngle switch
+                {
+                    >= 315 or < 45 => "Up",
+                    >= 45 and < 135 => "Left",
+                    >= 135 and < 225 => "Down",
+                    _ => "Right"
+                };
+                break;
+            case "Right":
+            case "Left":
+                itemDirection = itemAngle switch
+                {
+                    >= 315 or < 45 => "Down",
+                    >= 45 and < 135 => "Right",
+                    >= 135 and < 225 => "Up",
+                    _ => "Left"
+                };
+                break;
         }
-        return xform.LocalRotation.Degrees switch
+
+        return itemDirection switch
         {
-            < 45f =>  (int) DrawDepth.DrawDepth.Doors,
-            <= 315f =>  (int) DrawDepth.DrawDepth.WallMountedItems,
-            _ =>  (int) DrawDepth.DrawDepth.Doors,
+            "Up" => component.NorthOver ? (int) DrawDepth.DrawDepth.Doors : (int) DrawDepth.DrawDepth.WallMountedItems,
+            "Down" => component.SouthOver
+                ? (int) DrawDepth.DrawDepth.Doors
+                : (int) DrawDepth.DrawDepth.WallMountedItems,
+            "Left" => component.WestOver ? (int) DrawDepth.DrawDepth.Doors : (int) DrawDepth.DrawDepth.WallMountedItems,
+            "Right" => component.EastOver
+                ? (int) DrawDepth.DrawDepth.Doors
+                : (int) DrawDepth.DrawDepth.WallMountedItems,
+            _ => (int) DrawDepth.DrawDepth.Doors
         };
     }
 
+    private static int GetAngle(double degrees)
+    {
```

Comment:
use the built in `Angle` struct we have instead of manually determining it like this

Reply:
I did not find the function I needed there to get an angle from 0 to 360 from a degree of 4500 (turn the camera to the left many times).

Reply:
Ah, now I figured it out and did it.